string message_code
uint64 counter
uint16 hardware_id
uint16 value_num
exception/KeyValue[] values